### PR TITLE
Task Parallelism Detection

### PR DESCRIPTION
**Pull request by lukasrothenberger**
Merge request to include the current state of the task pattern detection into the develop branch.

### CODE REVIEW

File: discopop_explorer/PETGraphX.py [hunk 1]
```diff
@@ -173,53 +193,46 @@ class PETGraphX(object):
     g: nx.MultiDiGraph
     reduction_vars: List[Dict[str, str]]
     main: CUNode
-    pos: Dict
 
-    def __init__(self, g: nx.MultiDiGraph, reduction_vars: List[Dict[str, str]], pos):
-        self.g = g
+    def __init__(self, cu_dict: Dict[str, ObjectifiedElement], dependencies_list: List[DependenceItem],
```
Comment (aandergr) [inline discopop_explorer/PETGraphX.py:197]: Here (and the following lines) it seems to revert the change from b9102d891e7a05805232f536fa16e1ad11396d67. Why?

Comment (lukasrothenberger) [reply discopop_explorer/PETGraphX.py:197]: You´re right. Sorry. I´ll revert This change.

Comment (lukasrothenberger) [reply discopop_explorer/PETGraphX.py]: Should be fixed now.
